sdk/formrecognizer/azure-ai-formrecognizer/src/samples/java/com/azure/ai/formrecognizer/FormRecognizerAsyncClientJavaDocCodeSnippets.java

```diff
@@ -274,16 +329,68 @@ public void beginRecognizeReceiptsFromUrl() {
     }
 
     /**
-     * Code snippet for {@link FormRecognizerAsyncClient#beginRecognizeReceipts(Flux, long, FormContentType)}
+     * Code snippet for {@link FormRecognizerAsyncClient#beginRecognizeReceiptsFromUrl(String, RecognizeOptions)}
+     */
+    public void beginRecognizeReceiptsFromUrlWithOptions() {
+        // BEGIN: com.azure.ai.formrecognizer.FormRecognizerAsyncClient.beginRecognizeReceiptsFromUrl#string-recognizeOptions
+        String receiptUrl = "{file_source_url}";
+        boolean includeTextContent = true;
+        formRecognizerAsyncClient.beginRecognizeReceiptsFromUrl(receiptUrl,
+            new RecognizeOptions()
+                .setIncludeTextContent(includeTextContent)
+                .setPollInterval(Duration.ofSeconds(5)))
+            .subscribe(recognizePollingOperation -> {
+                // if training polling operation completed, retrieve the final result.
+                recognizePollingOperation.getFinalResult().subscribe(recognizedReceipts -> {
+                    for (int i = 0; i < recognizedReceipts.size(); i++) {
+                        RecognizedReceipt recognizedReceipt = recognizedReceipts.get(i);
```

Comment:
`subscribe` inside another `subscribe` looks a bit odd. Can we instead use reactor pattern here?

```java
formRecognizerAsyncClient.beginRecognizeReceiptsFromUrl(...)
        .flatMap(recognizePollingOperation -> recognizePollingOperation.getFinalResult())
        .subscribe(recognizedReceipts -> {...});
```